packages/json-machete/src/dereferenceObject.ts

```diff
@@ -8,7 +8,7 @@ export const resolvePath = (path: string, root: any): any => {
   return JsonPointer.get(root, path);
 };
 function isRefObject(obj: any): obj is { $ref: string } {
-  return typeof obj === 'object' && obj.$ref;
+  return typeof obj === 'object' && obj && obj.$ref;
```

Comment:
```suggestion
  return obj && typeof obj === 'object' && obj.$ref;
```
Is better as if `obj` is `null` you don’t assert `typeof obj === 'object'` condition